templates/clusterclass/nmt-cp.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: NutanixMachineTemplate
+metadata:
+  name: "${CLUSTER_CLASS_NAME}-cp-nmt"
+spec:
+  template:
+    spec:
+      providerID: "nutanix://${CLUSTER_NAME}-m1"
+      # Supported options for boot type: legacy and uefi
+      # Defaults to legacy if not set
+      bootType: ${NUTANIX_MACHINE_BOOT_TYPE=legacy}
```

Comment:
I agree with Jimmi, in general these templates should be very minimal, almost empty. And not expect env vars substitution here but have everything driven through patches and vars in `ClusterClass` and then all substitution happen at the `Cluster` level.
This keeps it extremally flexible where end users can reuse different levels:
1. All the templates but with a custom CC definition
2. Templates, CC and custom Cluster object
3. Templates, CC and sample Cluster for that easy `clusterctl` experience.

With that said I think the approach warrants a separate discussion/PR to align, so as long the tests are working we can always improve it in a follow up.

Reply:
resolving the comment here but let's get to the bottom of the feedback over successive PRs. @deepakm-ntnx